otter/convergence/model.py

```diff
@@ -112,6 +117,14 @@ class StepResult(Names):
     """
 
 
+@sumtype
+class ErrorReason(object):
+    """A reason for a step to be in a RETRY or FAILURE state."""
+    Exception = constructor('exc_info')
+    Other = constructor(reason=attr.ib(validator=instance_of((unicode, str))))
```

Comment:
Yeah, that makes sense. At first I was thinking it would allow arbitrary objects, but I guess I forgot about that idea when I added the str/unicode constraint :) I'll rename it to String.